### PR TITLE
Fixed issue #56

### PR DESCRIPTION
**Pull request by r1walz**

### CODE REVIEW

File: app/src/main/java/com/sdsmdg/kd/trianglifyexample/MainActivity.java [hunk 1]
```diff
@@ -38,6 +39,7 @@
     private CheckBox randomColoringCheckbox;
     private CheckBox customPaletteCheckbox;
     private Palette customPalette;
+    private final int PERMISSION_CODE = 69;
     public static final String PALETTE_COLOR_ARRAY = "Palette Color Array";
```
Comment (suyashmahar) [inline app/src/main/java/com/sdsmdg/kd/trianglifyexample/MainActivity.java]: Please move this to `strings.xml`.

File: app/src/main/java/com/sdsmdg/kd/trianglifyexample/MainActivity.java [hunk 2]
```diff
@@ -268,7 +270,13 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
 
     private void exportImage() {
         // Checks if permission is required for android version > 6
-        if (askForWritePermission() == 0) {
+        if (ContextCompat.checkSelfPermission(this,
```
Comment (suyashmahar) [inline app/src/main/java/com/sdsmdg/kd/trianglifyexample/MainActivity.java]: Nitpicking but something like the following would greatly increase readability:
```java
boolean permissionStatus = 
    ContextCompat.checkSelfPermission(this, Manifest.permission.WRITE_EXTERNAL_STORAGE);
if (permissionStatus != ...) {
```

**Review by suyashmahar — CHANGES_REQUESTED** [review]
Please upload an apk to drive and share a link here.

**Comment by r1walz** [comment]
APK: https://goo.gl/AjBeMV

**Pull request merged**